src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/HiddenFieldCheck.java

```diff
@@ -267,7 +263,18 @@ private void visitOtherTokens(DetailAST ast, int type) {
         else {
             frameName = null;
         }
-        final FieldFrame newFrame = new FieldFrame(frame, isStaticInnerType, frameName);
```

Comment:
Why not update conditions on `isStaticInnerType`?

Reply:
That is also fine. I am updating.

Reply:
Done